From 9364db9bfb3b9cbb331dd8a4d383de2135ea9ecd Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Wed, 20 May 2015 18:01:00 +0800
Subject: [PATCH 0316/5242] MLK-11004-1: ASoC: cs42888: Add regcache mask
 dirty in suspend

commit  d04e861aa54cac4e83de65f3147de625a7ab1989 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add regcache mask dirty in suspend, otherwise the register is not
synced with cache in resume.

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
(cherry picked from commit 9687aa5bbf5dd91b57f6e8b4b8ce69bf20e29c43)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/codecs/cs42xx8.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/sound/soc/codecs/cs42xx8.c b/sound/soc/codecs/cs42xx8.c
index c886f5882..0db214a 100644
--- a/sound/soc/codecs/cs42xx8.c
+++ b/sound/soc/codecs/cs42xx8.c
@@ -650,6 +650,7 @@ static int cs42xx8_runtime_suspend(struct device *dev)
 {
 	struct cs42xx8_priv *cs42xx8 = dev_get_drvdata(dev);
 
+	regcache_mark_dirty(cs42xx8->regmap);
 	regcache_cache_only(cs42xx8->regmap, true);
 
 	regulator_bulk_disable(ARRAY_SIZE(cs42xx8->supplies),
-- 
1.7.9.5

